src/app/digital/models/sim/Propagators.ts

```diff
@@ -32,12 +32,14 @@ export const AllPropagators: PropagatorRecord = {
     "DigitalNode": ({ signals }) => [{ "outputs": signals["inputs"] }],
 
     // Switch has state which represents the user-defined isOn/isOff
-    "Switch": ({ state = [Signal.Off] }) => [{ "outputs": state }, state],
+    "Switch": ({ state = [Signal.Off ] }) => [{ "outputs": state }, state],
 
     // LEDs don't propagate a signal
     "LED": Noprop,
 
     "ANDGate": ({ signals }) => [{ "outputs": [signals["inputs"].reduce(AND)] }],
+
+    "NOTGate": ({ signals }) => [{ "outputs": [...signals["inputs"]].map((x) => (Signal.fromBool(Signal.isOff(x)))) }],
```

Comment:
So the thing is that we know that `NOTGate` can only ever exactly 1 input, so we don't actually need to treat `"inputs"` as an array since we know it always has exactly 1 element.

Another useful thing we can do is add a `Signal.flip` function in `src/app/digital/models/sim/Signal.ts`, which will flip a single signal with the following conditions:
if the input signal is on, return Signal.Off
if the input signal is off, return Signal.On
if the input signal is Metastable, return Signal.Metastable

which you can then use here to hopefully have a very clean propagation function